src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -316,6 +327,100 @@ fn suggest_restriction(
 }
 
 impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
+    /// Try to find possible types that would satisfy the bounds in the type param to give an
+    /// appropriate turbofish suggestion.
+    fn get_turbofish_suggestions(
+        &self,
+        obligation: &PredicateObligation<'tcx>,
+        data: ty::TraitPredicate<'tcx>,
+        self_ty: Ty<'tcx>,
+    ) -> Vec<String> {
+        let mut turbofish_suggestions = FxHashSet::default();
+        self.tcx.for_each_relevant_impl(data.trait_ref.def_id, self_ty, |impl_def_id| {
+            let param_env = ty::ParamEnv::empty();
+            let param_env = param_env.subst(self.tcx, data.trait_ref.substs);
+            let ty = self.next_ty_var(TypeVariableOrigin {
+                kind: TypeVariableOriginKind::NormalizeProjectionType,
```

Comment:
This looks arbitrary, why not dummy? (or do we not have that anymore?)

Reply:
It was arbitrary, had changed it to `MiscVariable` in my local branch. Now it is moot using `probe`.